source/features/reactions-avatars.tsx

```diff
@@ -49,31 +48,31 @@ function getParticipants(container: HTMLElement): Participant[] {
 }
 
 function add(): void {
-	for (const list of select.all('.has-reactions .comment-reactions-options:not(.rgh-reactions)')) {
-		const avatarLimit = arbitraryAvatarLimit - (list.children.length * approximateHeaderLength);
+	for (const list of select.all('.has-reactions:not(.rgh-reactions) .comment-reactions-options')) {
+		const rows = list.children.length >= 4 ? 2 : 1;
+		const columnCount = Math.ceil(list.children.length / rows);
+		const avatarTotalCount = Math.ceil(approximateAvatarCount / columnCount);
```

Comment:
```suggestion
		const avatarReactionLimit = Math.ceil(avatarRowLimit / columnCount);
```

Also, I think these 3 values are unchanged in the whole loop, they can be moved out of it.

Also how about this?

```js
const avatarRowLimit = 22;
```

```js
const rows = list.children.length >= 4 ? 2 : 1;
const columnCount = Math.ceil(list.children.length / rows);
const avatarReactionLimit = Math.round(avatarRowLimit / columnCount) - 1; // Notice the subtle changes
```

This gives us:
- 1 column: 21 avatars (only 10 exist anyway) (unchanged)
- 2 columns: 10 avatars each (instead of 8)
- 3 columns: 6 avatars each (unchanged)
- 4 columns: 4 avatars each (unchanged)

Reply:
`const avatarRowLimit = 22;` was too high to also fit the "Add your reaction` button.